infra/Pos/Network/Types.hs

```diff
@@ -83,25 +88,48 @@ data Topology kademlia =
     -- | All peers of the node have been statically configured
     --
     -- This is used for core and relay nodes
-    TopologyCore !StaticPeers !(Maybe kademlia)
+    TopologyCore {
+        topologyStaticPeers :: !StaticPeers
+      , topologyOptKademlia :: !(Maybe kademlia)
+      }
 
-  | TopologyRelay !StaticPeers !(Maybe kademlia)
+  | TopologyRelay {
+        topologyStaticPeers :: !StaticPeers
+      , topologyOptKademlia :: !(Maybe kademlia)
+      }
 
     -- | We discover our peers through DNS
     --
     -- This is used for behind-NAT nodes.
-  | TopologyBehindNAT !Valency !Fallbacks !(DnsDomains DNS.Domain)
+  | TopologyBehindNAT {
+        topologyValency        :: !Valency
+      , topologyFallbacks      :: !Fallbacks
+      , topologyDnsDomains     :: !(DnsDomains DNS.Domain)
+      , topologyOptMaxAhead    :: !(Maybe OQ.MaxAhead)
+      , topologyOptRateLimit   :: !(Maybe OQ.RateLimit)
+      , topologyOptMaxInFlight :: !(Maybe OQ.MaxInFlight)
```

Comment:
https://github.com/input-output-hk/cardano-sl/pull/1347 obsoletes these last 3 fields. Instead of including policy in topology, a separate policy yaml can be given.